Digital ocean setup

kayiwa opened the pull request:
- sets up browser-trix on digital ocean

kayiwa committed:
create a k8s digital ocean cluster and load balancer

kayiwa committed:
add boto3 for SPaces

kayiwa committed:
add spaces and mongodb tasks

kayiwa committed:
add loadbalancer IP

kayiwa committed:
add project where possible

kayiwa committed:
add kubectl created loadbalancer

kayiwa committed:
use upstream nginx ingress

kayiwa committed:
add steps to run this playbook

kayiwa committed:
use doctl instead

kayiwa committed:
add documentation

kayiwa committed:
add complex deploy

ikreymer commented on ansible/playbooks/do_setup.yml:
```diff
@@ -0,0 +1,242 @@
+  vars:
+    - droplet_size: "s-1vcpu-2gb"
+    - droplet_region: "sfo3"
+    - do_project: "{{ my_project | default('btrixcloud-test')}}"
```
Where would `my_project`, `my_space_name` come from, if not set via `-e` flag?
Perhaps we just have a base name, and then have `{{ name }}-k8s` and `{{ name }-db` for each of the different components?

Does it make sense to store this in the inventory system somehow as an alternative to -e?

ikreymer commented on ansible/playbooks/do_setup.yml:
```diff
@@ -0,0 +1,242 @@
+    - space_name: "{{ my_space_name | default('btrixcloud-test')}}"
+    - btrix_db: "{{ my_db | default('btrixcloud-test')}}"
+    - k8s_cluster: "{{ my_k8s | default('btrixcloud-test')}}"
+    - domain_name: "{{ my_domain | default('example.edu')}}"
```
For setting the `A` record, we probably want to split this to a `domain` and `subdomain` field, with `subdomain` defaulting to '@'?

For  setting the full domain in the app, will need to then combine subdomain + domain

ikreymer commented on ansible/playbooks/do_setup.yml:
```diff
@@ -0,0 +1,242 @@
+        data: "{{ lb_ip | default('omit')}}"
+      ignore_errors: true
+
+    - name: d_ocean | register the dns of your webrecorder
```
can remove this section (setting domain via doctl), if doing this via the playbook above?

ikreymer commented on ansible/playbooks/do_setup.yml:
```diff
@@ -0,0 +1,242 @@
+    #  vars_files:
+    # - "../group_vars/do/vault.yml"
+  vars:
+    - droplet_size: "s-1vcpu-2gb"
```
```suggestion
    - droplet_size: "s-1vcpu-2gb"
    - main_node_size: "s-1vcpu-2gb"
    - crawl_node_size: "c-4"
```
can then use both settings in the create command

kayiwa committed:
Merge branch 'main' into digital_ocean_setup

kayiwa committed:
Update ansible/playbooks/do_setup.yml

add node sizes

Co-authored-by: Ilya Kreymer <ikreymer@users.noreply.github.com>

kayiwa committed:
remove printf statements from loadbalancer

kayiwa commented on ansible/group_vars/do/main.yml:
```diff
@@ -0,0 +1,11 @@
+droplet_region: "sfo3"
+do_project: "btrixcloud-{{ my_project | default('project')}}"
+space_name: "btrixcloud-{{ my_space_name | default('space')}}"
+btrix_db: "btrixcloud-{{ my_db | default('db')}}"
```
`{{ space_name | default('btrixcloud') }}-bucket`
`{{ db_name | default('btrixcloud') }}-db`

ikreymer replied:
perhaps something like this to set project_name as variable:
```
project_name: {{ project_name | default('btrixcloud') }}
space_name: {{ space_name | project_name }}-bucket
db_name: {{ db_name | project_name }}-db
```

kayiwa commented on ansible/playbooks/do_setup.yml:
```diff
@@ -0,0 +1,184 @@
+
+    - name: d_ocean | print information about an existing DigitalOcean Kubernetes cluster
+      ansible.builtin.debug:
+        msg: "Cluster name is {{ my_cluster.data.name }}, ID is {{ my_cluster.data.id }}"
```
pass this to `doctl k8s cluster list-associated-resources {{ my_cluster.data.id }} --format LoadBalancers --output json`
register: json_output

kayiwa committed:
fix variables

kayiwa committed:
create dbs via doctl

kayiwa committed:
extract keys from array of dictionaries

kayiwa committed:
variable consistency

kayiwa committed:
remove more printf statements

if a user has multiple loadbalancers or databases, getting this out
automatically could be disastrous.
Have a second run of playbook with variables populated

kayiwa committed:
extract mongodb_uri

kayiwa committed:
add documentation on how to use digital ocean playbook

kayiwa committed:
add documentation

document why the playbook will periodically need to be run multiple
times
ignore the private file which will have mongodb credentials

ikreymer committed:
undo changes to chart/values.yaml

ikreymer committed:
do playbook:
- automated init of db, cluster, main domain + signing domain
- move all config options into main.yml
- add options for enabling using do container registry, signing
- generate helm values chart ./deploys/ directory, run once

ikreymer committed:
cleanup playbook
add setting cors task

ikreymer committed:
chart: update default authsign bucket, remove mongo health check (for now)

kayiwa committed:
use ansible fqrn

ikreymer committed:
switch to helm via command
subdomain: default to project_name, cleanup domain setup logic

kayiwa committed:
add deletion of database and k8s cluster

kayiwa committed:
Merge remote-tracking branch 'upstream/main' into digital_ocean_setup

kayiwa committed:
add db_destroy tag since never will need variables from other tasks

kayiwa committed:
remove k8s and space

kayiwa committed:
setup instance on Digital Ocean
